pkg/controller/remotemachineset/remotemachineset_controller.go

```diff
@@ -493,6 +493,10 @@ func fetchAvailabilityZones(client awsclient.Client, region string) ([]string, e
 func decodeAWSMachineProviderSpec(rawExt *runtime.RawExtension, scheme *runtime.Scheme) (*awsprovider.AWSMachineProviderConfig, error) {
 	codecFactory := serializer.NewCodecFactory(scheme)
 	decoder := codecFactory.UniversalDecoder(awsprovider.SchemeGroupVersion)
+	if rawExt == nil {
+		// If the machine set has no ProviderSpec, return an empty object.
+		return &awsprovider.AWSMachineProviderConfig{}, nil
```

Comment:
Do you want this to return an error now so that it is output as a warning log back in `syncMachineSets`?